src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,83 @@
+  const dispatch = useDispatch();
+  const currentTodo = useSelector((state: RootState) => state.currentTodo);
+  const { id, title, completed, userId } = currentTodo!;
+  const [userInfo, setUserInfo] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  useEffect(() => {
+    const findUser = async (newUser: number) => {
+      setIsLoading(true);
+      try {
+        const user = await getUser(newUser);
+
+        setUserInfo(user);
+      } catch {
+        dispatch(clearCurrentTodo());
+        alert(`Didn't find user info`);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    findUser(userId);
+  }, [userId, dispatch]);
+
+  const handleClickCloseModal = () => {
+    dispatch(clearCurrentTodo());
+  };
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
-
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+      {isLoading ? (
+        <Loader />
+      ) : (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              Todo #{id}
+            </div>
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={() => handleClickCloseModal()}
+            />
+          </header>
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">
-            fugiat veniam minus
-          </p>
+          <div className="modal-card-body">
+            <p className="block" data-cy="modal-title">
+              {title}
+            </p>
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
+            <p className="block" data-cy="modal-user">
+              {completed ? (
+                <strong className="has-text-success">Done</strong>
+              ) : (
+                <strong className="has-text-danger">Planned</strong>
+              )}
 
-            {/* For completed */}
-            <strong className="has-text-success">Done</strong>
-            {' by '}
-            <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-          </p>
+              {' by '}
+              <a href={`mailto:${userInfo?.email}`}>{userInfo?.name}</a>
```

Comment:
It's recommended to check if `userInfo` is not null before accessing its properties. It might be null if the API call fails, and accessing properties of null would cause an error.